common/Perf/Azure.Test.Perf/EventPerfTest.cs

```diff
@@ -0,0 +1,80 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Azure.Test.Perf
+{
+    public abstract class EventPerfTest<TOptions> : PerfTestBase<TOptions> where TOptions : PerfOptionsBase
+    {
+        private readonly Stopwatch _stopwatch = new Stopwatch();
+
+        private long _completedOperations;
+        public sealed override long CompletedOperations => _completedOperations;
+
+        public sealed override IList<TimeSpan> Latencies =>
+            throw new InvalidOperationException("EventPerfTest does not support Latencies");
+
+        public sealed override IList<TimeSpan> CorrectedLatencies =>
+            throw new InvalidOperationException("EventPerfTest does not support CorrectedLatencies");
+
+        public EventPerfTest(TOptions options) : base(options)
+        {
+            if (options.Latency)
+            {
+                throw new InvalidOperationException("EventPerfTest does not support the '--latency' option");
+            }
+
+            if (options.Rate.HasValue)
+            {
+                throw new InvalidOperationException("EventPerfTest does not support the '--rate' option");
+            }
+        }
+
+        protected void EventRaised()
```

Comment:
This seems to be a great approach when you've got a single dispatch scenario.   Any thoughts on how we'd normalize between the T1 batched approach and the T2 single dispatch?   This would also be a potential consideration if we're looking to compare languages.    

If we try to just offset the difference by using a batch size of 1... but that would actually represent different behavior because of how the prefetch queue is managed.   We'd need to keep the batch size equal to the `CacheEventCount` in the T2 processor and somehow equate that with the number of events that the other processors were dispatching (which will vary from call-to-call)

Reply:
I prefer the shorter name (just `EventRaised()`), but I can rename if you think this is unclear.

In terms of normalizing between different APIs in T1 and T2, I agree we need to solve this for EventHubs, but I don't think it will have any impact on the perf framework itself.  I also agree we should implement a T1 vs T2 comparison test before merging this to validate the design.  I can work on this with you offline.